### ISSUE TITLE
Fix the path to example scripts in twisted.names.test.test_examples

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7352]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)| @wallrj reported|
|-|-|
|Trac ID|trac#7352|
|Type|release blocker: regression|
|Created|2014-05-15 08:21:47Z|
|Branch|https://github.com/twisted/twisted/tree/names-examples-path-7352|

I notice that the DNS example tests are being skipped since the latest re-arrangement of the documentation structure in r42144.

```
ames.test.test_examples.DnsServiceTests.test_shebang ... [SKIPPED]
twisted.names.test.test_examples.DnsServiceTests.test_usageConsistency ... [SKIPPED]
twisted.names.test.test_examples.DnsServiceTests.test_usageConsistencyOnError ... [SKIPPED]
twisted.names.test.test_examples.GetHostByNameTests.test_shebang ... [SKIPPED]
twisted.names.test.test_examples.GetHostByNameTests.test_usageConsistency ... [SKIPPED]
twisted.names.test.test_examples.GetHostByNameTests.test_usageConsistencyOnError ... [SKIPPED]
twisted.names.test.test_examples.MultiReverseLookupTests.test_shebang ... [SKIPPED]
twisted.names.test.test_examples.MultiReverseLookupTests.test_usageConsistency ... [SKIPPED]
twisted.names.test.test_examples.MultiReverseLookupTests.test_usageConsistencyOnError ... [SKIPPED]
twisted.names.test.test_examples.TestDnsTests.test_shebang ... [SKIPPED]
twisted.names.test.test_examples.TestDnsTests.test_usageConsistency ... [SKIPPED]
twisted.names.test.test_examples.TestDnsTests.test_usageConsistencyOnError ... [SKIPPED]
```

<details><summary>Searchable metadata</summary>

```
trac-id__7352 7352
type__release_blocker__regression release blocker: regression
reporter__rwall rwall
priority__normal normal
milestone__None None
branch__branches_names_examples_path_7352 branches/names-examples-path-7352
branch_author__rwall rwall
status__closed closed
resolution__fixed fixed
component__names names
keywords__ 
time__1400142107000000 1400142107000000
changetime__1445217216921098 1445217216921098
version__None None
owner__ioggstream ioggstream
cc__ioggstream
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="ioggstream's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|ioggstream set owner to @wallrj|
|-|-|

Fixed:

1- if tests can't find example files, raise FailTest instead of SkipTest

2- fixed example directory path

https://github.com/twisted/twisted/pull/39

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_3"></a>|@wallrj set status to `assigned`|
|-|-|

Reviewing...

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_4"></a>|@wallrj commented|
|-|-|

(In [42897]) Branching to names-examples-path-7352.

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_5"></a>|@wallrj set owner to ioggstream<br>@wallrj set status to `new`|
|-|-|

Thanks ioggstream

Unfortunately our idea for failing the tests rather than skipping them, doesn't work, because some of the buildbot slaves (the debian builders) run the tests against a packaged version of twisted; where the examples are not installed along side the modules.

So I've reverted that change to SkipTest.

And the tests now pass.

So I'll merge.

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_6"></a>|@wallrj set status to `closed`|
|-|-|

(In [42902]) Merge names-examples-path-7352

Author: ioggstream
Reviewers: rwall
Fixes: [#7352](https://github.com/twisted/twisted/issues/7352)

Fix paths to examples in twisted names tests.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_7"></a>|@hawkowl commented|
|-|-|

Ticket retargeted after milestone deleted
